### PR TITLE
Add to campaign styles UI enhancements

### PR DESCRIPTION
**Pull request by clarkepaul**
Relates to issue https://github.com/silverstripe/silverstripe-framework/issues/6058

### CODE REVIEW

**Comment by clarkepaul** [comment]
I just remember this isn't rebased, so I will do that shortly.

**Comment by clarkepaul** [comment]
After rebasing I've noticed the file area add to campaign area need updating too. I possibly need to create an asset-admin branch.

**Comment by clarkepaul** [comment]
Requires AssetAdmin branch for add to campaign within the files area. https://github.com/silverstripe/silverstripe-asset-admin/pull/282

File: admin/client/src/components/Modal/Modal.scss [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  .chosen-results > li {
+    padding: $spacer-y/1.5 $spacer-x;
+  }
+
+  .disabled-result {
+    font-style: italic;
+
+    &::after {
+      font-style: normal;
+      content: "Already contains item";
```
Comment (tractorcow) [inline admin/client/src/components/Modal/Modal.scss]: question: `modal` is a generic term given to popups that block access to the underlying page. This CSS is scoped to `modal__dialog`, but it seems very specific to the `add to campaign` feature, which is really only a single-use-case for modals.

Is there a way that the add-to-campaign content could be separated from the generic modal styles?

Comment (tractorcow) [inline admin/client/src/components/Modal/Modal.scss]: Perhaps as a simple win, rename from modal to campaignmodal?

Comment (clarkepaul) [inline admin/client/src/components/Modal/Modal.scss]: I was hoping that any styles here could be used generically for other modals. When I create new patterns I would like to create them as a component and not just for one use case like "add to campaign". You are right though in saying that the styles at the bottom are campaign specific with the disabled result—although it doesn't actually work or do anything, at one stage they did work. 

Best I move just styles out. Feels weird deleting it even though they don't work.

**Comment by clarkepaul** [comment]
I've removed the specific styles for campaign admin so everything else should be generic to any modal.

**Comment by tractorcow** [comment]
It looks like you force-pushed over the fix I added to the linting issue.

**Comment by tractorcow** [comment]
I've re-pushed the fix and retested it. Everything is good to merge on green.

**Comment by tractorcow** [comment]
Behat tests run perfectly fine locally.

**Pull request merged**